interface.go

```diff
@@ -79,6 +79,11 @@ type Job interface {
 	// of queue.ID()
 	Lock(string) error
 	Unlock(string)
+
```

Comment:
Write tests

Reply:
the current test suite goes a long way to ensure that this patch won't regress in cases where locks don't have scope (and already ensure that the primary locking mechanism (i.e. that jobs only ever run once.) is sufficient, so even though the tests haven't changed, I think passing tests provide good signal here. 

There's a trivial case that I plan to write once I get a green patch that if I add scopes to a bunch of sleeping jobs and too many threads, that it'll not dispatch half of them, but those tests are always kind of terrible.